ssl/s3_pkt.c

```diff
@@ -670,7 +670,7 @@ int ssl3_write_bytes(SSL *s, int type, const void *buf_, int len)
      * promptly send beyond the end of the users buffer ... so we trap and
      * report the error in a way the user will notice
      */
-    if (len < tot) {
+    if ((len < tot) || ((tot != 0) && (len < (tot + s->s3->wnum)))) {
```

Comment:
This doesn't make sense to me, as `s->s3->wnum` is always 0. It is set to that a few lines above.